install-civs

```diff
@@ -1,181 +1,183 @@
 #!/bin/bash
 
 TestAndCreateDir() {
-    if [ ! -d $1 ]; then
-	#echo "Directory $1 does not exist. Creating it.";
-        mkdir $1;
-    fi
+  if [ ! -d "$1" ]; then
+    #echo "Directory $1 does not exist. Creating it.";
+    mkdir "$1";
+  fi
 }
 
 
+settings_file="$1"
+
 if test $# -ne 1
 then
-  echo "Usage: install-civs <settings-file>";
-  exit 1;
+  echo "-- WARNING: no settings file provided; falling back to environment variables";
+  echo "-- Usage: install-civs [settings-file]";
+else
+  # shellcheck source=/dev/null
+  . "$settings_file" || exit 1
```

Comment:
I assume this reading from the environment is to support the docker stuff.  As a general rule I doubt like to read security sensitive stuff from the environment since it's publicly readable. How about reading from stdin and allowing - to be used as the file name to signify that?

Reply:
it's no less secure than having the values in plaintext, is it?

Reply:
Any process can read any other process' environment variables; try `ps -E`

Whereas read permissions control who can use a plaintext file.

Reply:
Ah right, gotcha.

But yeah, for better or for worse, injecting variables into the environment is generally how Docker deployments work.  If that's a dealbreaker then probably none of my proposed changes are gonna work out, except for maybe just the script linting.